pysatSpaceWeather/instruments/ace_epam.py

```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+"""Supports ACE Electron, Proton, and Alpha Monitor data
+
+Properties
+----------
+platform
+    'ace' Advanced Composition Explorer
+name
+    'epam' Electron, Proton, and Alpha Monitor
+tag
+    - 'realtime' Real-time data from the Space Weather Prediction Center (SWPC)
+    - 'historic' Historic data from the SWPC
+inst_id
+    - ''
+
+Note
+----
+This is not the ACE scientific data set, which will be available at pysatNASA
+
+The real-time data is stored by generation date, where each file contains the
+data for the current day.  If you leave download dates empty, though, it will
+grab today's file three times and assign dates from yesterday, today, and
+tomorrow.
+::
+
+    epam = pysat.Instrument('ace', 'epam', tag='realtime')
+    epam.download(start=epam.today())
+    epam.load(date=epam.today())
+
+
+
+Warnings
+--------
+The real-time data should not be used with the data padding option available
+from pysat.Instrument objects.
+
+"""
+
+import datetime as dt
+import functools
+import numpy as np
+
+from pysat import logger
+
+from pysatSpaceWeather.instruments.methods import ace as mm_ace
+
+# ----------------------------------------------------------------------------
+# Instrument attributes
+
+platform = 'ace'
+name = 'epam'
+tags = {'realtime': 'Real-time data from the SWPC',
+        'historic': ' Historic data from the SWPC'}
+inst_ids = {inst_id: [tag for tag in tags.keys()] for inst_id in ['']}
+
+# Define today's date
+now = dt.datetime.utcnow()
+
+# ----------------------------------------------------------------------------
+# Instrument test attributes
+
+# Set test dates (first level: inst_id, second level: tag)
+_test_dates = {inst_id: {'realtime': dt.datetime(now.year, now.month, now.day),
+                         'historic': dt.datetime(2009, 1, 1)}
+               for inst_id in inst_ids.keys()}
+
+# ----------------------------------------------------------------------------
+# Instrument methods
+
+preprocess = mm_ace.preprocess
```

Comment:
The preprocess function is not acting as expected. The fill_value update for variables isn't sticking for some reason. I'm thinking a pysat reason.

<img width="369" alt="Screen Shot 2021-05-17 at 5 57 20 PM" src="https://user-images.githubusercontent.com/3595641/118566403-6cb8c780-b739-11eb-89d5-3efb5a4d78f6.png">

Reply:
Ok, I think I fixed this now.